extensions/mssql/src/objectManagement/ui/S3AddBackupFileDialog.ts

```diff
@@ -49,7 +49,8 @@ export class S3AddBackupFileDialog extends DialogBase<S3AddBackupFileDialogResul
 		this.connectionUri = connectionUri;
 
 		this.dialogObject.okButton.onClick(async () => {
-			this.result.backupFilePath = `s3://${this.bucketDropdown.value}.s3.${this.regionInputBox.value}.amazonaws.com/${this.backupFilesDropdown.value}`;
+			// s3 objects can contain special characters that may cause issues with http request
+			this.result.backupFilePath = `s3://${this.bucketDropdown.value}.s3.${this.regionInputBox.value}.amazonaws.com/${encodeURIComponent(this.backupFilesDropdown.value.toString())}`;
```

Comment:
are `bucketDropdown` and `regionInputBox` guaranteed to not have special characters or should they also be wrapped in `encodeURIComponent()`?

Reply:
yup, I'll encode everything as suggested from Cory's comment just so it's easier to read, and also if they later change the limitations for buckets and regions.